src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -206,10 +207,14 @@ public void SendFunctionLoadRequests()
             }
         }
 
-        public void SendFunctionEnvironmentReloadRequest()
+        public void SendFunctionEnvironmentReloadRequest(TaskCompletionSource<bool> reloadTask)
         {
-            _eventSubscriptions.Add(_inboundWorkerEvents.Where(msg => msg.MessageType == MsgType.FunctionEnvironmentReloadResponse)
-      .Subscribe((msg) => PublishWorkerProcessReadyEvent(msg.Message.FunctionEnvironmentReloadResponse)));
+            _workerChannelLogger.LogDebug("Sending FunctionEnvironmentReloadRequest");
```

Comment:
This design assumes a single call and doesn't enforce that. The completion should be scoped to this request and managed by this type.

Reply:
thanks. Makes sense. Pushed an update